src/core/src/main/java/org/apache/jmeter/gui/action/AbstractAction.java

```diff
@@ -102,7 +102,6 @@ protected boolean popupCheckExistingFileListener(HashTree tree) {
                             return false;
                         }
                     case ASK:
-                    default:
```

Comment:
The removal of `default fo enums` is indeed the most controversial changes here.
However, in this case, the switch goes over `private enum`:

```java
    private enum ActionOnFile {
        APPEND,
        DELETE,
        ASK
    }
```

So it is clear there are only three possibilities, so the default can be omitted.

At the same time, if some of enum cases are missing, the compilation would fail.

Reply:
The questionable part is whether we trust enums from third-party classes. In other words, even if we cover all the current cases, it might be the library would add a new case, so if users upgrade the library without recompiling jmeter, then it won't detect "unexpected enum value".

WDYT? Is it really that important?
Frankly speaking, I incline that exhaustive switch over enum values looks better than arbitrary default that might unexpectedly catch newly added enum value. In other words, if we omit default for enums, then the newly added value would fail the compilation and ask the developer to add an appropriate case rather than falling into an unknown default.